lib/okta/jwk.ex

```diff
@@ -0,0 +1,32 @@
+defmodule Okta.Jwk do
+  def fetch_jwk(%{"kid" => key_id}, payload) do
+    with {:ok, jwks_uri} <- fetch_jwks_uri(payload),
+         {:ok, jwks_response} <- HTTPoison.get(jwks_uri),
+         {:ok, decoded_body} <- Jason.decode(jwks_response.body),
+         {:ok, keys} <- Map.fetch(decoded_body, "keys") do
+      {
+        :ok,
+        Enum.find(keys, fn key -> Map.fetch!(key, "kid") == key_id end)
```

Comment:
The JWKS endpoint may have several keys depending on the okta setup, return only the jwk with the key id specified in the header